### PR TITLE
align.AverageStructure for out-of-memory average structures

### PR DESCRIPTION
**Pull request by lilyminium**
Relates to #2039 

Changes made in this Pull Request:
 - added align.AverageStructure

PR Checklist
------------
 - [x] Tests?
 ~- [ ] Missing one to check the ``msf`` -- anyone have any existing tools to compute MSF?~ removed MSF
 - [x] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

### CODE REVIEW

File: package/MDAnalysis/analysis/align.py [hunk 1]
```diff
@@ -709,6 +709,200 @@ def save(self, rmsdfile):
         np.savetxt(rmsdfile, self.rmsd)
         logger.info("Wrote RMSD timeseries  to file %r", rmsdfile)
 
+class AverageStructure(AnalysisBase):
```
Comment (orbeckst) [inline package/MDAnalysis/analysis/align.py]: add duecredit for DOI 10.1021/acs.jpcb.7b11988

File: package/MDAnalysis/analysis/align.py [hunk 2]
```diff
@@ -709,6 +709,200 @@ def save(self, rmsdfile):
...
+
+        import MDAnalysis as mda
+        from MDAnalysis.tests.datafiles import PSF, DCD
+        from MDAnalysis.analysis import align
+
+        u = mda.Universe(PSF, DCD)
+
+        # align to the third frame and average structure
+        av = align.AverageStructure(u, ref_frame=3).run()
+        averaged_universe = av.universe
```
Comment (orbeckst) [inline package/MDAnalysis/analysis/align.py]: Cite the paper DOI https://doi.org/10.1021/acs.jpcb.7b11988 ; we typically put all references into references.rst and link to there. This is less convenient for `help(class)` but better for the online docs. If you want to also add a References section and don't link to the local section.

Comment (orbeckst) [reply package/MDAnalysis/analysis/align.py]: As far as I can tell, `gmx trjconv` has been able to do an iterative fit for a long time: http://manual.gromacs.org/documentation/2020/onlinehelp/gmx-trjconv.html with `-fit progressive`:

> There are two options for fitting the trajectory to a reference either for essential dynamics analysis, etc. The first option is just plain fitting to a reference structure in the structure file. The second option is a progressive fit in which the first timeframe is fitted to the reference structure in the structure file to obtain and each subsequent timeframe is fitted to the previously fitted structure. This way a continuous trajectory is generated, which might not be the case when using the regular fit method, e.g. when your protein undergoes large conformational transitions.

If `AverageStructure` does the same as `-fit progressive` then it should be mentioned.

@kain88-de – did you ever look at `gmx trjconv -fit progressive`?

Comment (lilyminium) [reply package/MDAnalysis/analysis/align.py]: @orbeckst Right now the class doesn't actually implement @kain88-de 's iterative fit from the paper, although it is heavily based off his `AverageStructure` class. I would suggest the AlignTraj class is a better fit for implementing `-fit progressive`.

Comment (orbeckst) [reply package/MDAnalysis/analysis/align.py]: Ah – I misunderstood.

File: package/MDAnalysis/analysis/align.py (hunk 1 above)
Comment (orbeckst) [reply package/MDAnalysis/analysis/align.py]: not needed

File: package/MDAnalysis/analysis/align.py (hunk 2 above)
Comment (kain88-de) [reply package/MDAnalysis/analysis/align.py]: @orbeckst I did not look into the gmx progressive fit. From the description, it is not identical to my algorithm though. This PR makes the implementation of the iterative algorithm a lot easier though. I updated the issue accordingly.

**Pull request merged**